python/paddle/distributed/collective.py

```diff
@@ -365,14 +367,60 @@ def _set_custom_gid(gid):
     _custom_gid = gid
 
 
-def new_group(ranks=None, backend=None):
+def _store_barrier(store, global_rank, global_world_size, timeout, group_name):
+    start_barrier_prefix = "StartBarrier/" + group_name + "/"
+    exit_barrier_key = "CanExitBarrier/" + group_name
+    end_barrier_prefix = "EndBarrier/" + group_name + "/"
+    is_master = global_rank == 0
+
+    def _check_keys_ready(wait_keys):
+        start_time = time.time()
+        while len(wait_keys) > 0:
+            time.sleep(0.5)
```

Comment:
Initialization of 2048 groups needs 2048/60/2=17 min?

Reply:
done